### ISSUE TITLE
Paste in new document causes black document

### ISSUE THREAD

**Issue by bladewolf55** [RickStrahl/MarkdownMonster#865]
**Description**  
Opening a new document and pasting text into it results in a black, blank page.

**Repro Steps**  
MM Preview is OFF
1.  Copy some text elsewhere
2. Open MM
3. Ctrl+N new document
4. Ctrl+V to paste

Expected: View pasted text
Actual: Page blacks out.

![image](https://user-images.githubusercontent.com/2903411/137172122-0b62b613-aa16-4fc8-b629-ad2599beb097.png)

**More Information**  
I can get the text to display by resizing the window or toggling the preview window twice.

**Comment by RickStrahl** [comment]
I don't see this here. What are you pasting into the doc? 

Have you tried this with different things? Wondering whether the problem is due to the new auto-content-dectection failing perhaps.

Maybe before anything update to v2.1 since 2.0.18.3 was around the time the auto-detection was added and there've been a few tweaks since.

**Comment by bladewolf55** [comment]
Same problem in 2.1
Animated GIFs attached.

It looks like pasting into a new document activates (or attempts to activate) the preview pane. 

When I activated preview in a new document, closed the document, then tried the same test the paste worked as expected.

BUT, when I then deactivated the preview, closed the document and tried the same test, pasting the text caused the preview pane to unexpectedly appear.

![2021-10-13_11-42-08](https://user-images.githubusercontent.com/2903411/137185878-05a8483c-9ee2-4d72-af3b-b8d13db3bae9.gif)
![2021-10-13_11-47-37](https://user-images.githubusercontent.com/2903411/137186094-efb3d270-9fc4-486e-97f9-a277c1466970.gif)

**Comment by RickStrahl** [comment]
Honestly I don't think this has anything to do with the pasting. It's about opening the document and the document not properly getting loaded it looks like.

Can you give me your exact steps? I have

* Open MM
* Click *New Document*
* Paste

Alternately using same steps with `Ctrl-n` same behavior.

[New Document.md](https://github.com/RickStrahl/MarkdownMonster/files/7341061/New.Document.md)




This works for me.

**Comment by RickStrahl** [comment]
Any update on this with more recent versions (2.1.5) - there have been a few more updates on initial page rendering since the last update on this post.

But I've not been able to repro this in any way even with extensive loading/unloading/re-loading and toggling the preview on and off in the process.

I do see the very, very rare scneario where tabs open with an empty, non-rendered preview, which I have not been able to reproduce. Activating another tab, then back fixes it but not been able to make it happen on demand so don't know how to fix. It's very rare though and getting even rarer with more recent releases of the WV control from Microsoft.

I'm going to close this for now - you can re-open if you still see this and have a repeatable scenario.

Thanks for your help!

**Comment by bladewolf55** [comment]
Thanks, Rick. Don't take this as a slam against MM, but I've switched to
using VS Code as my primary Markdown editor. It's not as conveniently
feature-rich, but doesn't have the issues I've been experiencing. One such
that I gave up on seeing fixed is the terrible lag time in MM when
activating snippets via keywords. In my journal--which is only a 600kB
file--I'd routinely experience a 5 second delay for each of my first two
snippets. With VS Code, it's type-tab-done.

What I'll miss is the ability to quickly write and run sophisticated C#
commands. And I've really appreciated your quick responses to filed issues.

--
Charles L Flatt
"If you're an advanced student and you're bored, you're not an advanced
student." --Jeff Hudson

**Comment by RickStrahl** [comment]
Hi charles 

Fair enough.

The snippets issue may be fixed in the latest update - I did some work on that in the last couple of days. The initial snippet lag time is for compiled snippets only. But there was a UI issue related to the recent async changes that I believe is now fixed. 

A 600kb Markdown file? I wouldn't consider that a small file :smile: 

Anyway - I hope you'll check back at some point. I'm fixing issues as they come. Unfortunately the move to the Chromium WebView has been more painful with some issues that took longer to find and work out.
